docker.local/bin/build.base.sh

```diff
@@ -10,9 +10,7 @@ do
     case $arg in
         -m1|--m1|m1)
```

Comment:
Is it possible to detect cpu architecture automatically instead of argument pass?

Reply:
Yeah! you pointed out a very valid point. 
docker should have detected the architecture and give the build appropriately. 
Why do we even need to pass `-m1` flag and handle it separately?

Reply:
We were handling it separately because of a known issue in the docker 

> Not all images are available for ARM64 architecture. You can add --platform linux/amd64 to run an Intel image under emulation. 

https://docs.docker.com/docker-for-mac/apple-silicon/

Reply:
But I'm able to build and run the image in `M1` in `arm64`. So I don't think so we need to do this special handling.

Reply:
it is good if you can test and get them worked on your Mac :)

if it is possible, `./docker.local/bin/build.base.sh` is better than `./docker.local/bin/build.base.sh -m1`

Reply:
https://cutecoder.org/software/detecting-apple-silicon-shell-script/

Reply:
At the stage when I was adding this flag there weren't Docker support of architecture whatsoever.
@NoSkillGuy @cnlangzi

Reply:
I've been testing without m1 flag now with new Docker 3.6.0 and new Mac Big Sur updates

Reply:
Version 3.5.0

<img width="901" alt="image" src="https://user-images.githubusercontent.com/805238/129207244-b83e50a2-b488-4d21-ad2f-cbdb44fe8a7d.png">

Reply:
> I've been testing without m1 flag now with new Docker 3.6.0 and new Mac Big Sur updates

Same here, I am working from past 3-4 days without passing `-m1` on M1 MacOS Big Sur and Docker Desktop Version 3.5.2 (3.5.2.18). I didn't get any issue :)